core/user-flow.js

```diff
@@ -67,11 +68,21 @@ class UserFlow {
   }
 
   /**
-   * @param {LH.UserFlow.StepFlags} [flags]
+   * @param {LH.UserFlow.StepFlags|undefined} flags
+   * @return {LH.UserFlow.StepFlags|undefined}
+   */
+  _getNextFlags(flags) {
+    const clonedFlowFlags = this._options?.flags && deepClone(this._options?.flags);
+    if (!flags) return clonedFlowFlags;
+    return mergeConfigFragment(clonedFlowFlags || {}, flags, true);
+  }
+
+  /**
+   * @param {LH.UserFlow.StepFlags|undefined} flags
    * @return {LH.UserFlow.StepFlags}
    */
   _getNextNavigationFlags(flags) {
-    const newStepFlags = {...flags};
+    const newStepFlags = this._getNextFlags(flags) || {};
```

Comment:
nit: change name, or put `|| {}` inside _getNextFlags

Reply:
I wanted `_getNextFlags` to return `undefined` if `undefined` is passed in.

`_getNextNavigationFlags` will never return `undefined` because we ensure that `skipAboutBlank` is set.